### PR TITLE
Add stm32f429zi board to the targets.

### PR DESCRIPTION
**Pull request by knightburton**
Completely new target for STM32F429ZI-Doscovery board based on K64F target.

### CODE REVIEW

File: targets/mbedstm32f429i/source/jerry_extapi.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    jerry_api_release_object (global_obj_p);
+    return false;
+  }
+
+  jerry_api_acquire_object (reg_func_p);
+  reg_value.type = JERRY_API_DATA_TYPE_OBJECT;
+  reg_value.u.v_object = reg_func_p;
+
+  bok = jerry_api_set_object_field_value (global_obj_p,
+                                          (jerry_api_char_t *)name,
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/jerry_extapi.cpp]: missing space after `)`

File: targets/mbedstm32f429i/source/jerry_extapi.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#define __JERRY_EXTAPI_H__
+
+#define JERRY_STANDALONE_EXIT_CODE_OK   (0)
+#define JERRY_STANDALONE_EXIT_CODE_FAIL (1)
+
+#define API_DATA_IS_OBJECT(val_p) \
+    ((val_p)->type == JERRY_API_DATA_TYPE_OBJECT)
+
+#define API_DATA_IS_FUNCTION(val_p) \
+    (API_DATA_IS_OBJECT(val_p) && \
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/jerry_extapi.h]: missing space before `(`

File: targets/mbedstm32f429i/source/jerry_extapi.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    jerry_api_is_function((val_p)->u.v_object))
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/jerry_extapi.h]: ditto

File: targets/mbedstm32f429i/source/jerry_run.cpp [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+int js_loop (uint32_t ticknow)
+{
+  jerry_api_object_t *global_obj_p;
+  jerry_api_value_t sysloop_func;
+  bool is_ok;
+
+  global_obj_p = jerry_api_get_global ();
+  is_ok = jerry_api_get_object_field_value (global_obj_p,
+                          (const jerry_api_char_t*)fn_sys_loop_name,
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/jerry_run.cpp]: missing space after `)`

File: targets/mbedstm32f429i/source/jerry_run.cpp [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    jerry_api_release_value (&sysloop_func);
+    jerry_api_release_object (global_obj_p);
+    return -2;
+  }
+
+  jerry_api_value_t* val_args;
+  uint16_t val_argv;
+
+  val_argv = 1;
+  val_args = (jerry_api_value_t*)malloc (sizeof (jerry_api_value_t) * val_argv);
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/jerry_run.cpp]: `(jerry_api_value_t *) malloc...`

File: targets/mbedstm32f429i/source/jerry_run.cpp [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  jerry_api_release_value (&sysloop_func);
+  jerry_api_release_object (global_obj_p);
+
+  return 0;
+}
+
+void js_exit (void)
+{
+  jerry_cleanup ();
+}
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/jerry_run.cpp]: please add a new line at the end of file

File: targets/mbedstm32f429i/source/jerry_run.h [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+#ifndef __JERRY_RUN_H__
+#define __JERRY_RUN_H__
+
+int js_entry (const char *source_p, const size_t source_size);
+int js_eval (const char *source_p, const size_t source_size);
+int js_loop (uint32_t ticknow);
+void js_exit (void);
+
+#endif
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/jerry_run.h]: ditto

File: targets/mbedstm32f429i/source/main.cpp [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+  pc.printf ("\r\nJerryScript in mbed STM32F429ZI\r\n");
+  pc.printf ("   build  %s\r\n", jerry_build_date);
+  pc.printf ("   hash   %s\r\n", jerry_commit_hash);
+  pc.printf ("   branch %s\r\n", jerry_branch_name);
+  
+  if (jerry_init() == 0)
+  {
+     minar::Scheduler::postCallback(jerry_loop).period(minar::milliseconds(100));
+	}
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/main.cpp]: wrong indentation

File: targets/mbedstm32f429i/source/main.cpp [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    {
+      pc.printf("js_eval failed code(%d) [%s]\r\n", retcode, js_codes[src].name);
+      js_exit();
+      return -2;
+    }
+  }
+  return 0;
+}
+
+static void jerry_loop(void)
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/main.cpp]: missing space

File: targets/mbedstm32f429i/source/main.cpp [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+static int jerry_init(void)
+{
+  int retcode;
+  int src;
+
+  DECLARE_JS_CODES;
+
+  /* run main.js */
+  retcode = js_entry(js_codes[0].source, js_codes[0].length);
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/main.cpp]: missing space

File: targets/mbedstm32f429i/source/main.cpp [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+#include "mbed-drivers/mbed.h"
+
+#include "jerry-core/jerry.h"
+#include "jerry_run.h"
+
+#include "jerry_targetjs.h"
+
+static Serial pc(USBTX, USBRX); //tx, rx
+
+static int jerry_init(void)
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/main.cpp]: missing space

File: targets/mbedstm32f429i/source/main.cpp [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+  /* run main.js */
+  retcode = js_entry(js_codes[0].source, js_codes[0].length);
+  if (retcode != 0)
+  {
+    pc.printf("js_entry failed code(%d) [%s]\r\n", retcode, js_codes[0].name);
+    js_exit();
+    return -1;
+  }
+  /* run rest of the js files */
+  for (src=1; js_codes[src].source; src++)
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/main.cpp]: please add spaces to assignment: `src = 1`

File: targets/mbedstm32f429i/source/main.cpp [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+static void jerry_loop(void)
+{
+  static uint32_t _jcount = 0;
+
+  js_loop(_jcount++);
+}
+
+void app_start(int, char**)
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/main.cpp]: missing space

File: targets/mbedstm32f429i/source/main.cpp [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+{
+  // set 9600 baud rate for stdout
+  pc.baud(9600);
+
+  pc.printf ("\r\nJerryScript in mbed STM32F429ZI\r\n");
+  pc.printf ("   build  %s\r\n", jerry_build_date);
+  pc.printf ("   hash   %s\r\n", jerry_commit_hash);
+  pc.printf ("   branch %s\r\n", jerry_branch_name);
+  
+  if (jerry_init() == 0)
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/main.cpp]: ditto

File: targets/mbedstm32f429i/source/makejerry.cmake [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+# compile flags
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mlittle-endian -mthumb -mcpu=cortex-m4" )
+
+# link jerryscript
+set(LJPATH ${CMAKE_CURRENT_LIST_DIR}/../libjerry)
+set(LJFILES "")
+set(LJFILES ${LJFILES} ${LJPATH}/libfdlibm.a)
+set(LJFILES ${LJFILES} ${LJPATH}/libjerrycore.a)
+target_link_libraries(${MBEDMODULE} ${LJFILES})
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/makejerry.cmake]: missing newline

File: targets/mbedstm32f429i/source/native_mbedstm32f429i.cpp [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+#include "mbed-drivers/mbed.h"
+
+#include "native_mbedstm32f429i.h"
+
+void native_led(int port, int val)
+{
+  	static const PinName portmap[] = { LED1, LED2, LED3, LED4 };
+  	static DigitalOut led(portmap[port]);
+  	led = val;
+}
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/native_mbedstm32f429i.cpp]: ditto

Comment (LaszloLango) [inline targets/mbedstm32f429i/source/native_mbedstm32f429i.cpp]: please don't use tab characters

File: targets/mbedstm32f429i/source/native_mbedstm32f429i.h [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NATIVE_MBEDSTM32F429I_H__
+#define __NATIVE_MBEDSTM32F429I_H__
+
+void native_led(int port, int val);
+
+#endif
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/native_mbedstm32f429i.h]: missing newline

File: targets/mbedstm32f429i/source/native_mbedstm32f429i.h [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NATIVE_MBEDSTM32F429I_H__
+#define __NATIVE_MBEDSTM32F429I_H__
+
+void native_led(int port, int val);
```
Comment (LaszloLango) [inline targets/mbedstm32f429i/source/native_mbedstm32f429i.h]: missing space before `(`

**Comment by polaroi8d** [comment]
In this and some other PRs what we did, they are way too much clones of each other, so this is the reason why I'm closed this pull request, because we created a new PR ( #1019 ) where merged the mbed releated targets into one source.

**Comment by knightburton** [comment]
In this and some other PRs what we did, they are way too much clones of each other, so this is the reason why I'm closed this pull request, because we created a new PR ( #1019 ) where merged the mbed releated targets into one source.

**Pull request closed without merging**